internal/armadactl/submit.go

```diff
@@ -14,44 +14,39 @@ import (
 
 // Submit a job, represented by a file, to the Armada server.
 // If dry-run is true, the job file is validated but not submitted.
-func (a *App) Submit(path string, dryRun bool) (outerErr error) {
+func (a *App) Submit(path string, dryRun bool) error {
 
 	ok, err := validation.ValidateSubmitFile(path)
 	if !ok {
-		fmt.Fprintf(a.Out, "Invalid jobfile: %s/n", err)
-		return nil
+		return err
 	}
 
 	submitFile := &domain.JobSubmitFile{}
-
 	err = util.BindJsonOrYaml(path, submitFile)
 	if err != nil {
-		return fmt.Errorf("[armadactl.Submit] error parsing job file: %s", err)
+		return err
 	}
 
 	if dryRun {
-		return
+		return nil
 	}
 
 	requests := client.CreateChunkedSubmitRequests(submitFile.Queue, submitFile.JobSetId, submitFile.Jobs)
-	client.WithConnection(a.Params.ApiConnectionDetails, func(conn *grpc.ClientConn) {
-		submissionClient := api.NewSubmitClient(conn)
+	return client.WithSubmitClient(a.Params.ApiConnectionDetails, func(c api.SubmitClient) error {
 		for _, request := range requests {
-			response, err := client.SubmitJobs(submissionClient, request)
+			response, err := client.SubmitJobs(c, request)
 			if err != nil {
-				outerErr = fmt.Errorf("[armadactl.Submit] error submitting job with request %#v: %s", request, err)
-				return
+				return errors.WithMessagef(err, "error submitting request %#v", request)
```

Comment:
Just looked at docs for [WithMessage](https://pkg.go.dev/github.com/pkg/errors#hdr-Adding_context_to_an_error), and honestly I do not understand when to use `Wrap` and when to use `WithMessage`

Reply:
Wrap adds both a message and a stack trace.